### PR TITLE
Simplified some booleans and if elses

### PR DESCRIPTION
**Pull request by ApamNapat**
I'm positive that this doesn't change the way code works but clearly reduces boilerplate and redundancy

### CODE REVIEW

**Comment by inigo** [comment]
Those changes all look correct to me.

**Comment by phyzical** [comment]
i understand the reasoning for the changes, but i think this pattern is done so that the "last comparison" return doesn't cause issues if any new ones need to be added after it, as in if the developer needs a new "last comparison" they just add it before the default return. if im making any sense 😆

File: src/data/codingcontracttypes.ts [hunk 1]
```diff
@@ -307,15 +307,7 @@ export const codingContractTypesMetadata: ICodingContractTypeMetadata[] = [
...
-
-      if (ans === "1" && solution) {
-        return true;
-      }
-      if (ans === "0" && !solution) {
-        return true;
-      }
-
-      return false;
+      return ans === "1" && solution || ans === "0" && !solution;
```
Comment (phyzical) [inline src/data/codingcontracttypes.ts:310]: shouldn't we add parenthesis for better readability

**Comment by ApamNapat** [comment]
> i understand the reasoning for the changes, but i think this pattern is done so that the "last comparison" return doesn't cause issues if any new ones need to be added after it, as in if the developer needs a new "last comparison" they just add it before the default return. if im making any sense 😆

This applies to some cases only, but anyways I think this is easy to revert if needed (YAGNI kinda thing) and I avoided the changes for cases where there were long `if {} else if {} ...` chains for the reasons you gave

File: src/data/codingcontracttypes.ts (hunk 1 above)
Comment (ApamNapat) [reply src/data/codingcontracttypes.ts:310]: You convinced me me with JS doing weird stuff

**Comment by phyzical** [comment]
nah i agree proably should have said its more of a `nit pick comment :)`

File: src/ExportBonus.tsx [hunk 2]
```diff
@@ -6,8 +6,7 @@ export let LastExportBonus = 0;
 const bonusTimer = 24 * 60 * 60 * 1000; // 24h
 export function canGetBonus(): boolean {
   const now = new Date().getTime();
-  if (now - LastExportBonus > bonusTimer) return true;
-  return false;
+  return now - LastExportBonus > bonusTimer;
```
Comment (phyzical) [inline src/ExportBonus.tsx:9]: parentheses for this one too :)

Comment (ApamNapat) [reply src/ExportBonus.tsx:9]: Now this one really doesn't convince me

**Review by phyzical — APPROVED** [review]
lgtm

**Pull request merged**